src/services/IGameContext.hh

```diff
@@ -0,0 +1,42 @@
+#ifndef RA_SERVICES_IGAMECONTEXT_HH
+#define RA_SERVICES_IGAMECONTEXT_HH
+#pragma once
+
+#include <string>
+
+namespace ra {
+namespace services {
+
+class IGameContext
+{
+public:
```

Comment:
You definitely need a pure virtual destructor. 

The rest would be nice to put it in.
I'd normally say save this for later but since the impl class seems like it's supposed to be ID'd.
Delete copying, then put in default moving and default construction as protected.

Normally you're supposed to delete copying anyway with abstract classes to eliminate splicing but the unique constraint that an ID implies seems to warrant this more. To save time you can just copy and paste this. I was thinking of doing this for all our classes later, except the ones that already have them implemented.

```
...
public:
  virtual ~IGameContext() noexcept = default;
  IGameContext(const IGameContext&) noexcept = delete;
  IGameContext& operator=(const IGameContext&) noexcept = delete;
...
protected:
  IGameContext() noexcept = default;
  IGameContext(IGameContext&&) noexcept = default;
  IGameContext& operator=(IGameContext&&) noexcept = default;
...
```

Then you can override the non-deleted ones in implementations if you need non-default behavior.
Some people say you can just make the ID const and it won't change, that is a myth, it's still possible for duplicate entries to be made resulting in exceptions being thrown. Plus it won't be move assignable. Deleting copying will guarantee it won't be duplicated.

Reply:
As a service, it shouldn't be moveable either. I've added the suggested constructors/destructors with the move functions also marked as deleted.